### PR TITLE
Fix issue accepting regional RM invite

### PR DESCRIPTION
**Pull request by LaurelOlson**
## Description

Fixes an issue where a logged in reviewer/review manger accepts a regional RM invitation sent to a different email than the one they are logged in with. Previously, this was upgrading the user to a regional RM but since the regional RM was not invited with the proper jurisdictions, the user only got assigned to their existing jurisdiction.
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents

[BPHH-1736](https://hous-bssb.atlassian.net/browse/BPHH-1736)

## Steps to QA

1. login as reviewer or review manager
2. login as admin in separate session. Invite a regional RM with an email that is NOT currently assigned to a review staff.
3. Accept the invite in the browser where the reviewer/review manager is logged in. After accepting the invite, the user should be come a regional RM and should have both the original jurisdiction and the jurisdiction the regional RM was invited to in the jurisdiction select.

## UI Accessibility Checklist

_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

- [ ] Markup uses semantic HTML?
- [ ] Additional context added through `aria-roles` and `aria-labels`?
- [ ] Checked
      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?
- [ ] Usable with a keyboard and navigable in a logical order?
- [ ] Usable using a screen reader (e.g. Voiceover for Mac) with enough context for the user?

## General Checklist

- [ ] ✅ Provide tests for your changes where relevant
- [x] 📝 Use descriptive commit messages
- [ ] 📗 Update any related documentation and include any relevant screenshots
- [ ] 📱 For UI-related tasks, ensure responsive design is implemented across multiple screen size
- [x] 🗂️ Move your relevant story/task to the _Ready for Code Review_ state

## [optional] Are there any post-deployment tasks we need to perform?

n/a

### CODE REVIEW

File: app/services/jurisdiction/user_inviter.rb [hunk 1]
```diff
@@ -37,8 +37,8 @@ def invite_users
           user.invite!(inviter)
           self.results[:reinvited] << user
         elsif inviter.invitable_roles.include?(user_params[:role].to_s)
-          jurisdiction_id = user_params[:jurisdiction_id] || inviter.jurisdiction&.id
-          user = User.new(user_params.merge({ discarded_at: nil, jurisdiction_id: }))
+          jurisdiction_id = user_params.delete(:jurisdiction_id) || inviter.jurisdictions.pluck(:id)
+          user = User.new(user_params.merge({ discarded_at: nil, jurisdiction_ids: [jurisdiction_id].flatten }))
```
Comment (taw27) [inline app/services/jurisdiction/user_inviter.rb:41]: wouldn't `inviter.jurisdictions.pluck(:id)` be an array of ids? Will this work if this is the case?

File: db/schema.rb [hunk 2]
```diff
@@ -64,12 +64,6 @@
             name: "index_contacts_on_contactable"
   end
 
-  create_table "data_migrations",
-               primary_key: "version",
-               id: :string,
-               force: :cascade do |t|
-  end
-
```
Comment (taw27) [inline db/schema.rb:72]: This gets auto generated for the `rails data:migrate`, I am assuming it's not there if you haven't run local data migrations.  Please run rails data:migrate to put this back

**Review by taw27 — CHANGES_REQUESTED** [review]

File: db/schema.rb (hunk 2 above)
Comment (LaurelOlson) [reply db/schema.rb:72]: @taw27 weird - I did a DB dump to make sure it was in line and this got removed.. What version of PG are you on?

Comment (LaurelOlson) [reply db/schema.rb:72]: nvm I misread that, will do

Comment (taw27) [reply db/schema.rb:72]: @LaurelOlson data migrations are separate from db:migrations and don't get run automatically (intentionally). If you run rails data migrate or rails db:migrate:with_data this table will get generated along with the data migrations (which isn't relevant for clean db)

File: app/services/jurisdiction/user_inviter.rb (hunk 1 above)
Comment (LaurelOlson) [reply app/services/jurisdiction/user_inviter.rb:41]: that's what the `.flatten` is for

Comment (taw27) [reply app/services/jurisdiction/user_inviter.rb:41]: @LaurelOlson 
Makes sense, didn't notice the flatten

File: db/schema.rb (hunk 2 above)
Comment (LaurelOlson) [reply db/schema.rb]: OK fixed, thanks!

**Review by taw27 — APPROVED** [review]

**Pull request merged**